hypnotoad/core/equilibrium.py

```diff
@@ -764,9 +794,43 @@ class PsiContour:
     Mostly behaves like a list
     """
 
-    options = Options(refine_width=1.0e-5, refine_atol=2.0e-8, refine_methods="line")
+    # options = Options(refine_width=1.0e-5, refine_atol=2.0e-8, refine_methods="line")
+    user_options_factory = OptionsFactory(
+        # Include settings for member FineContour objects
+        FineContour.user_options_factory.defaults,
+        refine_width=WithMeta(
+            1.0e-5,
+            doc="width for line search when refining points",
+            value_type=float,
+            checks=is_positive,
+        ),
+        refine_atol=WithMeta(
+            2.0e-8,
+            doc="absolute tolerance for refinement of points",
+            value_type=float,
+            checks=is_positive,
+        ),
+        refine_methods=WithMeta(
+            "integrate+newton, integrate",
+            doc=(
+                """
+                ordered, comma-separated list of methods to try when refining points:
+                Valid names are:
+                - "newton"       Newton iteration
+                - "line"         A line search
+                - "integrate"    Integrate along psi gradient
+                - "integrate+newton"  Integrate, then refine with Newton
+                - "none"         No refinement (always succeeds)
+            """
+            ),
+            value_type=str,
+            # Don't check 'allowed' here, because we also support any permutation of
+            # these options, and correctness of value is checked where it is used.
+            # allowed=["newton", "line", "integrate", "integrate+newton", "none"],
+        ),
+    )
 
-    def __init__(self, points, psi, psival):
+    def __init__(self, *, points, psi, psival, settings):
```

Comment:
Yes, that's what it does. I like this pattern when the argument list starts getting long because you get a clear error if you make a mistake with the required arguments.